src/NLog/MappedDiagnosticsContext.cs

```diff
@@ -63,20 +63,39 @@ public static void Set(string item, string value)
         }
 
         /// <summary>
+        /// Sets the current thread MDC item to the specified value.
+        /// </summary>
+        /// <param name="item">Item name.</param>
+        /// <param name="value">Item value.</param>
+        public static void Set(string item, object value)
+        {
+            ThreadDictionary[item] = value;
+        }
+
+        /// <summary>
         /// Gets the current thread MDC named item.
         /// </summary>
         /// <param name="item">Item name.</param>
         /// <returns>The item value of string.Empty if the value is not present.</returns>
         public static string Get(string item)
         {
-            string s;
+            object o = GetObject(item);
+            return (o != null) ? o.ToString() : String.Empty;
```

Comment:
I missed this when adding RenderContextObject. It has been corrected to use ConvertToString now.